src/main/java/org/ice4j/stack/StunClientTransaction.java

```diff
@@ -486,27 +301,21 @@ void cancel()
      */
     public void handleResponse(StunMessageEvent evt)
     {
-        lock.lock();
-        try
-        {
-            TransactionID transactionID = getTransactionID();
-
-            logger.log(Level.FINE, "handleResponse tid " + transactionID);
-            if(!Boolean.getBoolean(StackProperties.KEEP_CRANS_AFTER_A_RESPONSE))
-                cancel();
-
-            responseCollector.processResponse(
-                    new StunResponseEvent(
-                            stackCallback,
-                            evt.getRawMessage(),
-                            (Response) evt.getMessage(),
-                            request,
-                            transactionID));
-        }
-        finally
+        TransactionID transactionID = getTransactionID();
+
+        logger.log(Level.FINE, "handleResponse tid " + transactionID);
+        if(!Boolean.getBoolean(StackProperties.KEEP_CRANS_AFTER_A_RESPONSE))
         {
-            lock.unlock();
+            cancel();
```

Comment:
surprised we don't early return here, but i see the original code didn't so i guess best to keep it that way

Reply:
@bbaldino it looks like this method is kind of "interceptor" of responses, what it does - it stops re-transmission if it is configured to stop on receiving response and it passed response to actual handler who cares about it.

Reply:
yup, i see now